web/src/main/java/com/epam/rft/atsy/web/controllers/LoginController.java

```diff
@@ -10,16 +15,29 @@
  */
 @Controller
 @RequestMapping("/login")
-public class LoginController {
-  private static final String VIEW_NAME = "login";
+public class LoginController implements ApplicationContextAware {
+  private static final String VIEW_NAME_LOGIN = "login";
+  private static final String VIEW_NAME_WELCOME = "secure/welcome";
+
+  @Autowired
+  AuthenticationService authenticationService;
 
   /**
    * Loads the page.
    * @return the ModelAndView object which describes the page
    */
   @RequestMapping(method = RequestMethod.GET)
   public ModelAndView pageLoad() {
-    return new ModelAndView(VIEW_NAME);
+
+    if (this.authenticationService.isCurrentUserAuthenticated()) {
+      return new ModelAndView("redirect:/" + VIEW_NAME_WELCOME);
+    } else {
+      return new ModelAndView(VIEW_NAME_LOGIN);
+    }
   }
 
-}
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
```

Comment:
This method is not required now as the `@Autowired` works when running the application, and Mockito's `@InjectMocks` injects the mocked AuthenticationService in the `LoginControllerTest`.